### PR TITLE
Add unit test for icon.py

### PR DESCRIPTION
**Pull request by anthonyabeo**
I improved the test coverage by adding test instances for the icon.py file in core/toga/widgets. Specifically, l added coverage for lines 40-45 and 56.

## PR Checklist:

- [x] All new features have been tested
- [ ] All new features have been documented
- [x ] I have read the **CONTRIBUTING.md** file
- [x ] I will abide by the code of conduct

### CODE REVIEW

File: src/core/toga/widgets/icon.py [hunk 1]
```diff
@@ -39,10 +39,11 @@ def __init__(self, path, system=False):
     def filename(self):
         if self.system:
             toga_dir = os.path.dirname(os.path.dirname(__file__))
+            print(toga_dir)
             return os.path.join(toga_dir, 'resources', self.path)
         else:
-            from toga.app import App
-            return os.path.join(App.app_dir, self.path)
+            # from toga.app import App
+            return os.path.join("", self.path)
```
Comment (freakboy3742) [inline src/core/toga/widgets/icon.py]: Is this change intentional? It looks like something that might have been commented out while you were exploring some aspect of the API.

File: src/core/toga/app.py [hunk 2]
```diff
@@ -47,6 +47,8 @@ class App:
         >>> app.main_loop()
     """
     app = None
+    app_dir = None
+    app_module = None
```
Comment (freakboy3742) [inline src/core/toga/app.py]: Why are these two needed? You should be able to confirm if `app_dir` has a value by checking if `app` exists

File: src/core/tests/widgets/test_icon.py [hunk 3]
```diff
@@ -11,10 +12,28 @@ def setUp(self):
         self.factory.Icon = MagicMock(return_value=MagicMock(spec=toga_dummy.factory.Icon))
 
         self.test_path = "Example.bmp"
+        self.system = False
```
Comment (freakboy3742) [inline src/core/tests/widgets/test_icon.py]: What is the purpose of this line?

File: src/core/toga/app.py (hunk 2 above)
Comment (anthonyabeo) [reply src/core/toga/app.py]: I was getting an error with the "App.app_dir".  It said App has no attribute "app_dir". I wasn't sure why the App.app_dir was doing since app_dir was only available when a new app is created.

File: src/core/tests/widgets/test_icon.py (hunk 3 above)
Comment (anthonyabeo) [reply src/core/tests/widgets/test_icon.py]: I was setting self.system to True and False to test for the instances where an icon is in the resources directory or not.

File: src/core/toga/widgets/icon.py (hunk 1 above)
Comment (anthonyabeo) [reply src/core/toga/widgets/icon.py]: Yes. It was intentional. I was getting an error with the "App.app_dir" because there was no instance of App, hence "app_dir" was not available.

File: src/core/tests/widgets/test_icon.py (hunk 3 above)
Comment (freakboy3742) [reply src/core/tests/widgets/test_icon.py]: Yes, but the value isn't actually *used* anywhere.

File: src/core/toga/widgets/icon.py (hunk 1 above)
Comment (freakboy3742) [reply src/core/toga/widgets/icon.py]: Right - so there is an underlying bug that needs to be fixed, not just papering over the cause.

Comment (anthonyabeo) [reply src/core/toga/widgets/icon.py]: Sure. I am just starting out with open source, so i am still new to some of these thing. But will keep that in mind for later.

File: src/core/tests/widgets/test_icon.py (hunk 3 above)
Comment (anthonyabeo) [reply src/core/tests/widgets/test_icon.py]: Yes, that's correct. The value of self.system is set to False by default. Hence, there was no need to explicitly add that line. It can be removed.

**Comment by anthonyabeo** [comment]
Hello @freakboy3742 , i just added a couple of test cases for splitcontainer.py. I noticed that for the 'set_app' and 'set_window' methods, the app and window parameters were not used in the method. Is this delibrate? figured you might want to check it out. Therefore, when i was writing the test for those methods, i simply set a default value of None for now. So that i can revisit it after i receive some clarifications.

**Comment by freakboy3742** [comment]
@anthonyabeo Yes, it is deliberate; the reasoning is explained in Ticket #299. There's a general pattern we're using in argument passing to backends, which will often result in the provided argument not being used.

**Comment by freakboy3742** [comment]
@anthonyabeo Also - there's not much point adding more tests to this PR until the existing code has been approved and merged. Adding more code just means more feedback, and the PR doesn't get any closer to being merged.

**Comment by anthonyabeo** [comment]
Sure @freakboy3742. The intent was to create a new PR, but somehow it got merged into the existing one.
